### DISCUSSION TITLE
Potential security vulnerability due to very old package versions in vendor/modules/code-oss-dev?

### DISCUSSION THREAD

**Discussion by ivanov-slk** [coder/code-server#4554] [Q&A]
Hi there,

I noticed that there are many outdated packages under `code-server-3.12.0-linux-amd64.tar/code-server-3.12.0-linux-amd64/vendor/modules/code-oss-dev`. `code-oss-dev` itself looks suspicious - it has been [reported as containing malicious software ](https://www.npmjs.com/package/code-oss-dev) on `npm`.

A screenshot of two of the outdated packages:
![image](https://user-images.githubusercontent.com/42289817/143591432-975933d9-ea78-4e54-900e-658c68943a2d.png)

The [latest version](https://www.npmjs.com/package/json) for `json` is 11.0.0.
For `handlebars` [it is 4.7.7](https://www.npmjs.com/package/handlebars).

These outdated packages are reported as vulnerable by the security scanner I'm using.

I'd like to know are these packages needed? How can they be updated? What about the security issue in `code-oss-dev`?

Thank you!

**Comment by lg-nineoneninethreenine** [comment]
@jsjoeio knows how the packages are needed, I think I sound offensive, so no offense. I think the old packages are used because that might be small in size and they are actually properly doing the job and also it could be that since the modules are updated, the code also might change, right? So maybe due to that, code-server uses old versions. If you can, try investigating more if the modules are upgradable and if they are, try opening a PR, looping in this discussion, maybe I'll also review(requires a build test, hope you have a good Mac or Linux machine)

**Comment by jsjoeio** [comment]
Hmm...great questions. `code-oss-dev` is the name of VS Code packaged up and used in code-server. Our fork lives [here](https://github.com/cdr/vscode).

We're in the process of getting a release ready for 4.0.0. 

If those packages exist in `code-oss-dev`, then they're probably needed by VS Code. We have [trivy](https://github.com/cdr/code-server/blob/main/.github/workflows/ci.yaml#L424-L443) and another script that checks for vulnerabilities [here](https://github.com/cdr/code-server/blob/main/ci/dev/audit.sh). Those should catch anything suspicious.
